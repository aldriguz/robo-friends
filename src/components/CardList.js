import React from 'react';
import Card from '../components/Card';

const CardList = ({ robots }) => {
    console.log('CardList');
    const cardsArray = robots.map((user, i) => {
        return <Card key={i} id={user.id} name={user.name} email={user.email}/>
    });

    return(
        <React.Fragment>
            {cardsArray}
        </React.Fragment>
    );
}

export default CardList;