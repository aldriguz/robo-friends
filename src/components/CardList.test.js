import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('test if Card component has been rendered', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'Jhon Snow',
            username: 'jhonsnow',
            email: 'jhon@email.com'
        }
    ]

    expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
})
