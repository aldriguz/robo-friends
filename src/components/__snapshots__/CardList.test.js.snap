// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if Card component has been rendered 1`] = `
<Fragment>
  <Card
    email="jhon@email.com"
    id={1}
    key="0"
    name="Jhon Snow"
  />
</Fragment>
`;
